lib/active_merchant/billing/gateways/shift4.rb

```diff
@@ -197,17 +198,16 @@ def add_card_present(post, options)
         post[:card][:present] = options[:card_present] || 'N'
       end
 
-      def add_customer(post, options)
-        if address = options[:billing_address]
-          post[:customer] = {}
-          post[:customer][:addressLine1] = address[:address1] if address[:address1]
-          post[:customer][:postalCode] = address[:zip]
-          name = address[:name].split(' ') if address[:name]
-          if name&.is_a?(Array)
-            post[:customer][:firstName] = name[0]
-            post[:customer][:lastName] = name[1]
-          end
-        end
+      def add_customer(post, options, card = nil)
```

Comment:
We should swap the `options` and `card` params to fit the AM method pattern. 

We don't need to set `card` too `nil` since in each of the transactions where this is referenced, a payment method is expected. Line 207 and 208 handle any issue where the card is not a `CreditCard`.

Reply:
Thanks @naashton , I have updated this change 👍